In [1]:
import pandas as pd

df2 = pd.read_csv("premier_player_info.csv")

df2.head()

player_image_url      player_name  \
0  https://resources.premierleague.com/premierlea...       Max Aarons   
1  https://resources.premierleague.com/premierlea...    George Abbott   
2  https://resources.premierleague.com/premierlea...      Zach Abbott   
3  https://resources.premierleague.com/premierlea...  Josh Acheampong   
4  https://resources.premierleague.com/premierlea...        Ché Adams   

  player_country        player_club player_position  \
0        England        Bournemouth        Defender   
1        England  Tottenham Hotspur      Midfielder   
2        England  Nottingham Forest        Defender   
3        England            Chelsea        Defender   
4       Scotland        Southampton         Forward   

                                    player_stats_url  
0  https://www.premierleague.com/en/players/23298...  
1  https://www.premierleague.com/en/players/51932...  
2  https://www.premierleague.com/en/players/54906...  
3  https://www.premierleague.com/en/players/57701...  
4  https://www.premierleague.com/en/players/20043...

In [3]:
cols = list(df2.columns)

In [4]:
cols.insert(0, cols.pop(1))

In [6]:
df2 = df2[cols]

Mettre player_name en première position

In [7]:
df2.head()

player_name                                   player_image_url  \
0       Max Aarons  https://resources.premierleague.com/premierlea...   
1    George Abbott  https://resources.premierleague.com/premierlea...   
2      Zach Abbott  https://resources.premierleague.com/premierlea...   
3  Josh Acheampong  https://resources.premierleague.com/premierlea...   
4        Ché Adams  https://resources.premierleague.com/premierlea...   

  player_country        player_club player_position  \
0        England        Bournemouth        Defender   
1        England  Tottenham Hotspur      Midfielder   
2        England  Nottingham Forest        Defender   
3        England            Chelsea        Defender   
4       Scotland        Southampton         Forward   

                                    player_stats_url  
0  https://www.premierleague.com/en/players/23298...  
1  https://www.premierleague.com/en/players/51932...  
2  https://www.premierleague.com/en/players/54906...  
3  https://www.premierleague.com/en/players/57701...  
4  https://www.premierleague.com/en/players/20043...

In [8]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1116 entries, 0 to 1115
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   player_name       1116 non-null   object
 1   player_image_url  1116 non-null   object
 2   player_country    1116 non-null   object
 3   player_club       1116 non-null   object
 4   player_position   1116 non-null   object
 5   player_stats_url  1116 non-null   object
dtypes: object(6)
memory usage: 52.4+ KB


Voir s'il y a le meme nombre de ligne que dans l'ancienne table (1116)

In [9]:
df2 = df2.query("player_name not in ['Freddie Symonds', 'Tom Taylor']").reset_index(drop=True)

In [10]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1114 entries, 0 to 1113
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   player_name       1114 non-null   object
 1   player_image_url  1114 non-null   object
 2   player_country    1114 non-null   object
 3   player_club       1114 non-null   object
 4   player_position   1114 non-null   object
 5   player_stats_url  1114 non-null   object
dtypes: object(6)
memory usage: 52.3+ KB


Supprimer les 2 joueurs supprimé de l'autre table

In [11]:
df2.to_pickle("df2_modifie.pkl")